src/Spryker/Zed/Shipment/Persistence/ShipmentRepository.php

```diff
@@ -33,4 +34,52 @@ public function isShipmentMethodUniqueForCarrier(ShipmentMethodTransfer $shipmen
             ->filterByFkShipmentCarrier($shipmentMethodTransfer->getFkShipmentCarrier())
             ->exists();
     }
+
+    /**
+     * @param string $shipmentMethodName
+     *
+     * @return \Generated\Shared\Transfer\ShipmentMethodTransfer|null
+     */
+    public function findShipmentMethodByName(string $shipmentMethodName): ?ShipmentMethodTransfer
+    {
+        $salesShipmentMethodEntity = $this->queryMethodsWithMethodPricesAndCarrier()
+            ->filterByName($shipmentMethodName)
+            ->find()
+            ->getFirst();
+
+        if ($salesShipmentMethodEntity === null) {
+            return null;
+        }
+
+        return $this->getFactory()
+            ->createShipmentMethodMapper()
+            ->mapShipmentMethodEntityToShipmentMethodTransferWithPrices(
+                $salesShipmentMethodEntity,
+                new ShipmentMethodTransfer()
+            );
+    }
+
+    /**
+     * @module Currency
+     *
+     * @return \Orm\Zed\Shipment\Persistence\SpyShipmentMethodQuery
+     */
+    protected function queryMethodsWithMethodPricesAndCarrier(): SpyShipmentMethodQuery
+    {
+        return $this->queryMethods()
+            ->joinWithShipmentMethodPrice()
+                ->useShipmentMethodPriceQuery()
+                    ->joinWithCurrency()
+                    ->joinWithStore()
```

Comment:
`spy_shipment_method_store` table is introduced only in 8.0.0
https://github.com/spryker/shipment/blob/8.0.0/src/Spryker/Zed/Shipment/Persistence/Propel/Schema/spy_shipment.schema.xml#L68

you cannot use it in the query.

See this method implementation in version 7.0.0